srcalc/src/main/java/gov/va/med/srcalc/vista/RpcVistaPatientDao.java

```diff
@@ -223,25 +225,69 @@ private void parseRecentVitalResults(final Patient patient, final List<String> v
         // For example, pulse is returned as: "Pulse:       (03/05/10@09:00)  74  _NURSE,ONE_Vitals"
         final SimpleDateFormat dateFormat = new SimpleDateFormat(
                 "(" + VISTA_DATE_OUTPUT_FORMAT.toPattern() + ")");
-        final Pattern compliedPattern = Pattern.compile(VITALS_SPLIT_REGEX);
-        // Each entry comes with an accompanying date and time.
-        final List<String> heightLineTokens = Splitter.on(compliedPattern).splitToList(vitalResults.get(5));
-        final int feet = Integer.parseInt(heightLineTokens.get(2));
-        patient.setHeight(new RetrievedValue(
-                (feet * 12.0) + Double.parseDouble(heightLineTokens.get(4)),
-                dateFormat.parse(heightLineTokens.get(1)),
-                HEIGHT_UNITS));
-        final List<String> weightLineTokens = Splitter.on(compliedPattern).splitToList(vitalResults.get(6));
-        patient.setWeight(new RetrievedValue(
-                Double.parseDouble(weightLineTokens.get(2)),
-                dateFormat.parse(weightLineTokens.get(1)),
-                WEIGHT_UNITS));
-        final List<String> bmiLineTokens = Splitter.on(compliedPattern).splitToList(vitalResults.get(7));
-        // The BMI value is the second to last token on its line
-        patient.setBmi(new RetrievedValue(
-            Double.parseDouble(bmiLineTokens.get(bmiLineTokens.size()-2)),
-            patient.getWeight().getMeasureDate(),
-            ""));
+        for(final String line: vitalResults)
+        {
+            // For each line, determine if we need to pull the vital information for that line
+            // Each entry comes with an optional date and time.
+            final Pattern pattern = Pattern.compile(
```

Comment:
You can move this pattern compilation outside of the for loop for an easy performance gain.